### ISSUE TITLE
Streaming by line inconsistencies

### ISSUE THREAD

**Issue by joshmeads** [yamafaktory/jql#70]
Hey @yamafaktory, first of all, thank you! JQL is fantastic

However, I've noticed issues when streaming logs into JQL. I've looked at #26 and tried every option I can think of but I'm getting zero output.

This works as expected:
```
docker logs container_name 2>&1 | jql '.!'
```

Fails when trying to tail via `docker logs -f`
```
docker logs -f container_name 2>&1 | jql '.!'
```

Right now I'm using JQ to process them, (`docker logs -f container_name 2>&1 | jq '.'`) but I honestly love the `'.!'` option.

Any ideas on how to make this work? Wondering if it's similar to many JQ comments suggesting you need `--unbuffered --stream`.

**Comment by yamafaktory** [comment]
Hi @joshmeads and thanks for the feedback!

At the moment `jql` doesn't support reading a real stream of JSON as input. In #26, what I've done was using `Deserializer::from_str` from `serde_json`, but this is only valid for a file.

However there's a super handy function which should make this possible: `Deserializer::from_reader` (https://docs.serde.rs/serde_json/fn.from_reader.html). I'd have to refactor the way the binary currently takes the data.

Bonus point: this will be unbuffered out of the box.

> The content of the IO stream is deserialized directly from the stream without being buffered in memory by serde_json.

Will soon have a look at it and keep you updated!

**Comment by yamafaktory** [comment]
@joshmeads I've just released v2.8.0 with a new `--stream` flag that should fix your issue.

Please feel free to reopen this issue if this is not working as expected.

Thanks again for the feedback!

https://github.com/yamafaktory/jql#streaming

**Comment by joshmeads** [comment]
@yamafaktory you're a legend. Works perfectly. Thank you so much!
